core/frontend/src/ViewCreator3d.ts

```diff
@@ -91,8 +91,17 @@ export class ViewCreator3d {
     const categories: Id64Array = await this._getAllCategories();
 
     // model extents
+    let modelExtents: Range3d = this._imodel.projectExtents;
     const modelProps = await this._imodel.models.queryModelRanges(models);
-    const modelExtents = Range3d.fromJSON(modelProps[0]);
+
+    if (modelProps.length > 0) {
+      modelExtents = new Range3d();
+      for (const props of modelProps)
+        modelExtents = modelExtents.union(Range3d.fromJSON(props));
+    }
+
+    if (modelExtents == null) modelExtents = this._imodel.projectExtents;

```

Comment:
The range will never be `null`. It may be a null range. Check `isNull` property.
Adhere to style guidelines - body of `if` statement goes on separate line.